src/pykiso/lib/connectors/trc_handler.py

```diff
@@ -0,0 +1,325 @@
+        """
+        type_ = cols[self.columns["T"]]
+        bus = self.columns.get("B", None)
+
+        # For 2.0 ST / ER and EC messages dlc and length are not in the trace but length is fixed
+        if type_ in ["EC", "ER", "ST"] and self.file_version == TRCFileVersion.V2_0:
+            if type_ == "ST":
+                length = 4
+            elif type_ == "ER":
+                length = 5
+            else:
+                length = 2
+            dlc = "  "
+        else:
+            if "l" in self.columns:
+                length = int(cols[self.columns["l"]])
+                dlc = len2dlc(length)
+            elif "L" in self.columns:
+                dlc = int(cols[self.columns["L"]])
+                length = dlc2len(dlc)
+            else:
+                raise ValueError("No length/dlc columns present.")
+
+        msg = Message()
+        if isinstance(self.start_time, datetime):
+            msg.timestamp = (
+                self.start_time + timedelta(milliseconds=float(cols[self.columns["O"]]))
+            ).timestamp()
+        else:
+            msg.timestamp = float(cols[1]) / 1000
+
+        if type_ not in ["EC", "ER", "ST"] or self.file_version == TRCFileVersion.V2_1:
+            msg.arbitration_id = int(cols[self.columns["I"]], 16)
+            msg.is_extended_id = len(cols[self.columns["I"]]) > 4
+        # No arbitration id for ST / ER and EC messages
+        else:
+            msg.arbitration_id = cols[self.columns["I"]]
+            msg.is_extended_id = False
+
+        msg.channel = int(cols[bus]) if bus is not None else 1
+        msg.dlc = dlc
+        msg.data = bytearray(
+            [int(cols[i + self.columns["D"]], 16) for i in range(length)]
+        )
+        msg.is_rx = cols[self.columns["d"]] == "Rx"
+        msg.is_fd = type_ in ["FD", "FB", "FE", "BI"]
+        msg.bitrate_switch = type_ in ["FB", " FE"]
+        msg.error_state_indicator = type_ in ["FE", "BI"]
+        msg.is_error_frame = type_ == "ER"
+        msg.is_remote_frame = type_ == "RR"
+
+        return TypedMessage(type_, msg)
+
+
+class TRCWriterCanFD(TRCWriter):
+    """Logs CAN FD data to text file (.trc)"""
+
+    # Type has been added to FORMAT_MESSAGE >= 2.0
+    FORMAT_MESSAGE_V2_1 = "{msgnr:>7} {time:13.3f} {type:>2} {channel:>2} {id:>8} {dir:>2} - {dlc:<4} {data}"
+
+    FORMAT_MESSAGE_V2_0 = (
+        "{msgnr:>7} {time:13.3f} {type:>2} {id:>8} {dir:>2} {dlc:<2} {data}"
+    )
+
+    def _format_message_init(self, msg: TypedMessage, channel: int) -> str:
+        """Pick message format from file version and format initial message
+
+        :return: message informations formated in a string
+        """
+        # Fix error from python can -> message number should start at one not zero
+        self.msgnr = 1
+
+        if self.file_version == TRCFileVersion.V1_0:
+            self._msg_fmt_string = self.FORMAT_MESSAGE_V1_0
+            self._format_message = self._format_message_by_format
```

Comment:
Wouldn't it be way more concise to just run these lines once at the very end of the function?
```python
self._format_message = self._format_message_by_format
return self._format_message_by_format(msg, channel)
```

Reply:
Done